core/rootchain.go

```diff
@@ -192,10 +167,23 @@ func (rc *RootChain) loopEvent(eventName string, event interface{}) {
 		}
 
 		for _, event := range events {
-			// filter repeat block
-			currentBlockEventData.DelLastBlockEventHash(event.BlockNum, rc.chainDb)
+			if currentBlockEventData.blockNum != 0 && currentBlockEventData.blockNum != event.BlockNum {
+				currentBlockEventData.DelLastBlockEventHash(rc.chainDb)
+				// save fromblock after finish one Block
+				err = rc.PutFromBlock(big.NewInt(int64(event.BlockNum)))
```

Comment:
should save `fromBlock` before `DelLastBlockEventHash`, as if the saving operation failed then no event will be processed twice next time (e.g. system recovered and starts to reprocessing the events)